src/main/java/org/jitsi/jibri/RecordingSinkType.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright @ 2018 Atlassian Pty Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.jibri;
+
+import org.codehaus.jackson.annotate.*;
+
+/**
+ * @author bbaldino
+ * NOTE: this file had to be in java, I couldn't get it to recognize the
+ * custom JsonCreator constructor to handle enum-case-agnosticisim in kotlin
```

Comment:
Could be due to the annotations not being picked up?

e.g. for our serialised types we use `@field:SerializedName("xyz")` so that it works with Java based annotation processors

Reply:
i didn't try that, i tried basically exactly what's there in java (using `@JsonCreator`) but when it was a kotlin file, for some reason it didn't pick it up.